hs_geographic_feature_resource/receivers.py

```diff
@@ -212,51 +212,65 @@ def geofeature_pre_create_resource(sender, **kwargs):
     tmp_dir = None
     try:
         files = kwargs['files']
-        metadata = kwargs['metadata']
-        validate_files_dict = kwargs['validate_files']
         fed_res_fnames = kwargs['fed_res_file_names']
         fed_res_path = kwargs['fed_res_path']
-        file_selected = False
-
-        if files:
-            file_selected = True
-            file_list = [file.name for file in files]
-        elif fed_res_fnames:
-            file_list = utils.get_fed_zone_files(fed_res_fnames)
-            if file_list:
-                file_selected = True
-
-        if file_selected:
-            files_type = check_uploaded_files_type(files, file_list)
+        metadata = kwargs['metadata']
+        validate_files_dict = kwargs['validate_files']
+
+        if files and fed_res_fnames:
+            validate_files_dict['are_files_valid'] = False
+            validate_files_dict['message'] = 'Please upload files from either local disk or irods, not both.'
+            return
+
+        if files or fed_res_fnames:
+            files_list = [f for f in files]
```

Comment:
@pkdash Yes, files is already a list, but as I can see here @zhiyuli wants to create a new list here to hold all files from files list as well as files coming from federated zone. @zhiyuli I do see a performance issue here. I see you put files downloaded from federated zone into UploadedFile() object and append them into the new list, which while makes the code simplier to handle, it does incur additional performance penalty especially for large files from federation zones. My original implementation is to try to avoid this performance penalty. We may need to chat offline if you have questions, but I think it'd be best to avoid putting files from federated zone into UploadedFile().